modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemsList.js

```diff
@@ -16,33 +16,37 @@ import ClayIcon from '@clayui/icon';
 import PropTypes from 'prop-types';
 import React, {useContext} from 'react';
 
-import Summary from '../summary/Summary';
-import CartItem from './CartItem';
-import CartItemsListActions from './CartItemsListActions';
 import MiniCartContext from './MiniCartContext';
-import {summaryDataMapper} from './util/index';
+import {ADD_PRODUCT} from './util/constants';
 
 function CartItemsList({items}) {
-	const {cartState, isUpdating, spritemap} = useContext(MiniCartContext),
+	const {
+			CartViews,
+			cartState,
+			isUpdating,
+			labels,
+			spritemap,
+			summaryDataMapper,
+		} = useContext(MiniCartContext),
 		{summary = {}} = cartState,
 		{itemsQuantity = 0} = summary,
-		numberOfItems = (items || []).length;
+		{length: numberOfItems = 0} = items || [];
```

Comment:
(Non-blocking feedback but) we're [going to add a lint](https://github.com/liferay/liferay-frontend-projects/issues/11) soon to enforce one statement per variable declaration, so you might want to move towards that style to reduce future churn (ie. one variable per `const` statement).

Out of curiosity, why the change from?

```js
numberOfItems = (items || []).length;
```

to?

```js
{length: numberOfItems = 0} = items || [];
```

It seems (subjectively) harder to read and (objectively) longer. Is it that `items` might be some exotic non-array that has an `undefined` `length` property?

Reply:
No exotic arrays, I've been the exotic one in this case, the only reason being that I wanted to put some "more evident" default `length: numberOfItems = 0`. I'll move to the more solid "one-statement-per-var-declaration" and take away some of these lines in favour of some optional chaining checks. 👍